internal/auth/providers/amazon_cognito_admin.go

```diff
@@ -0,0 +1,270 @@
+package providers
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/cognitoidentityprovider"
+	"github.com/buzzfeed/sso/internal/auth/circuit"
+	log "github.com/buzzfeed/sso/internal/pkg/logging"
+	"github.com/buzzfeed/sso/internal/pkg/sessions"
+	"github.com/datadog/datadog-go/statsd"
+)
+
+type CognitoAdminProvider interface {
+	ListMemberships(groupName string) (members []string, err error)
+	CheckMemberships(userName string) (inGroups []string, err error)
+	GlobalSignOut(session *sessions.SessionState) (err error)
+	GetUserInfo(accessToken string) (userInfo *cognitoidentityprovider.GetUserOutput, err error)
+}
+
+type CognitoAdminService struct {
+	adminService *cognitoidentityprovider.CognitoIdentityProvider
+	StatsdClient *statsd.Client
+	cb           *circuit.Breaker
+	userPoolID   *string
+}
+
+func getCognitoIdentityProvider(id, secret, region string) (*cognitoidentityprovider.CognitoIdentityProvider, error) {
+	logger := log.NewLogEntry()
+	sess, err := session.NewSession(&aws.Config{
```

Comment:
since we have our _own_ package named `session` -- it's probably worth adding an alias so it is explicit this is an aws session pkg. Through myself there for a loop first time I read this 😨

Reply:
Oh yeah...hah. Bet that was confusing at first 😬